packages/hub/src/scope.ts

```diff
@@ -470,7 +471,12 @@ export class Scope implements ScopeInterface {
     event.breadcrumbs = [...(event.breadcrumbs || []), ...this._breadcrumbs];
     event.breadcrumbs = event.breadcrumbs.length > 0 ? event.breadcrumbs : undefined;
 
-    event.sdkProcessingMetadata = this._sdkProcessingMetadata;
+    // Since we're storing dynamic sampling context data in the event.sdkProcessingMetadata
+    // field We have to re-apply it after we applied the Scope's field.
+    // (This is because we're storing this data on the span and not on the scope)
+    const baggage = event.sdkProcessingMetadata && event.sdkProcessingMetadata.baggage;
+    event.sdkProcessingMetadata = this._sdkProcessingMetadata || {};
+    event.sdkProcessingMetadata.baggage = baggage;
```

Comment:
> Did this line just overwrite/wipe the entire metadata from before? 

Yes, everything that was on `event.sdkProcessingMetadata`.

I had the same feeling when writing this hack. Decided to go with it anyway because I thought that there was a reason for why we do this. 
However, I just tried your suggestion and our tests still seem to pass, with it in, so I'll take it. Thanks :D